clustered/client/src/test/java/org/ehcache/clustered/client/internal/store/ReconnectingServerStoreProxyTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.client.internal.store;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.terracotta.exception.ConnectionClosedException;
+
+import java.nio.ByteBuffer;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.doThrow;
+
+public class ReconnectingServerStoreProxyTest {
+
+  @Mock
+  ServerStoreProxy proxy;
+
+  @Mock
+  Runnable runnable;
+
+  @Rule
+  public ExpectedException exception = ExpectedException.none();
+
+  private final ServerStoreProxyException storeProxyException = new ServerStoreProxyException(new ConnectionClosedException("Connection Closed"));
+
+  private ReconnectingServerStoreProxy serverStoreProxy;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
```

Comment:
If you want, you can use a MockitoRule and an `@InjectMocks` to do constructor injection. It will allow to remove the entire `setUp()`

Reply:
cool